operator/pkg/translate/translate.go

```diff
@@ -147,9 +148,27 @@ func (t *Translator) OverlayK8sSettings(yml string, iop *v1alpha1.IstioOperatorS
 		}
 		inPath = strings.Replace(inPath, "gressGateways.", "gressGateways."+fmt.Sprint(index)+".", 1)
 		scope.Debugf("Checking for path %s in IstioOperatorSpec", inPath)
-		m, found, err := tpath.GetFromStructPath(iop, inPath)
-		if err != nil {
-			return "", err
+
+		var found bool
+		var m interface{}
+		if componentName == name.AddonComponentName {
```

Comment:
it would be better to do this inside renderFeatureComponentPathTemplate as a special case template. this kind of fixup if going to be hard to read and maintain.

Reply:
I can create a helper function for this, would that be better? The reason is that we are not just changing the templates but there is also some logic changed required(due to the limitation of the tpath code we discussed offline)